lib/entice/logic/vitals.ex

```diff
@@ -21,44 +23,97 @@ defmodule Entice.Logic.Vitals do
   def unregister(entity_id),
   do: Entity.remove_behaviour(entity_id, Vitals.Behaviour)
 
+  #External API
+  def damage(entity, amount) do
+
+  end
 
   defmodule Behaviour do
     use Entice.Entity.Behaviour
     alias Entice.Logic.Vitals.Health
     alias Entice.Logic.Vitals.Energy
+    alias Entice.Logic.Vitals.Morale
     alias Entice.Logic.Player.Level
 
 
     def init(entity, _args) do
       {:ok, entity |> put_attribute(get_max_health(entity))
-                   |> put_attribute(get_max_energy(entity))}
+                   |> put_attribute(get_max_energy(entity))
+                   |> put_attribute(%Morale{morale: 0})}
+    end
 
+    def init(entity, [:entity_resurrected, percent_health, percent_energy]) do
+      {_, max_health: max_health} = get_max_health(entity)
+      resurrected_health = max_health / 100 * percent_health
+
+      {_, max_mana: max_mana} = get_max_energy(entity)
+      resurrected_mana = max_mana / 100 * percent_energy
+
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: resurrected_health, max_health: max_health} end)
+                   |> update_attribute(Energy, fn _ -> %Energy{mana: resurrected_mana, max_mana: max_mana} end)}
     end
 
+    def termiante(:remove_handler, entity) do
+      {:ok, entity}
+    end
 
     def terminate(_reason, entity) do
       {:ok, entity |> remove_attribute(Health)
                    |> remove_attribute(Energy)}
     end
 
+    def handle_event({:vitals_entity_damage, amount}, %Entity{id: id, attributes: %{Health => %Health{health: health, max_health: max_health}}} = entity) do
+      new_health = health - amount
+      cond do
+        new_health <= 0 -> {:become, Vitals.DeadBehaviour, [], entity |> update_attribute(Health, fn health -> %Health{health: 0, max_health: health.max_health} end)}
+        new_health > 0 -> {:ok, entity |> update_attribute(Health, fn health -> %Health{health: new_health, max_health: health.max_health} end)}
+      end
+    end
 
-    # internal
+    def handle_event({:vitals_entity_heal, amount}, %Entity{id: id} = entity) do
+      {:ok, %Health{health: health, max_health: max_health}} = fetch_attribute(entity, Health)
+      new_health = health + ammount
+      cond do
+        (health + amount) > max_health -> new_health = max_health
+        (health + amount) < max_health -> new_health = health + amount
+      end
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: new_health, max_health: max_health})}
+    end
 
+    # internal
 
     defp get_max_health(entity) do
```

Comment:
since we should have both of the following values in the init method (by match etc) we can pass them in directly instead of the entity state